src/databaseAdapters/season.ts

```diff
@@ -0,0 +1,70 @@
+import Season from '../models/season';
+import { ISeason } from '../models/types/season.types';
+
+interface DatabaseAdapter {
```

Comment:
Let's unify this. And add getters on the parent DatabaseAdapter to getUserAdapter() or something similar.